src/os/file.go

```diff
@@ -115,10 +116,23 @@ func (f *File) Read(b []byte) (n int, err error) {
 // It returns the number of bytes read and any error encountered, possible io.EOF.
 // At end of file, Read returns 0, io.EOF.
 func (f *File) ReadAt(b []byte, offset int64) (n int, err error) {
-	n, err = f.handle.ReadAt(b, offset)
-	if err != nil && err != io.EOF {
-		err = &PathError{"readat", f.name, err}
+	if offset < 0 {
+		return 0, &PathError{Op: "readat", Path: f.name, Err: errors.New("negative offset")}
+	}
+
+	for len(b) > 0 {
+		m, e := f.handle.ReadAt(b, offset)
```

Comment:
You could consider moving this to the implementation of `f.handle.ReadAt`. Not sure about that. It would make `f.handle.ReadAt` also follow the `io.ReaderAt` documentation:

> ReadAt reads len(p) bytes into p starting at offset off in the underlying input source. It returns the number of bytes read (0 <= n <= len(p)) and any error encountered.

Reply:
The loop upstream is in the `File` methods, not in the `handle` version: https://github.com/golang/go/blob/master/src/os/file.go#L127